internal/provider/validation.go

```diff
@@ -85,41 +86,55 @@ func validateYggdrasil(apiClient *apiClient, sub subi.SubstrateExt) error {
 	}()
 	c, err := net.Dial("tcp", fmt.Sprintf("%s:%d", listenIP, port))
 	if err != nil {
-		return errors.Wrapf(err, "failed to connect to ip. make sure the twin id is associated with a valid yggdrasil ip, twin id: %d, ygg ip: %s, err", apiClient.twin_id, yggIP)
+		return errors.Wrapf(err, "failed to connect to ip. make sure the twin id is associated with a valid yggdrasil ip, twin id: %d, ygg ip: %s, err", threefoldPluginClient.twinID, yggIP)
 	}
 	c.Close()
 	if !arrived {
-		return errors.Wrapf(err, "sent request but didn't arrive to me. make sure the twin id is associated with a valid yggdrasil ip, twin id: %d, ygg ip: %s, err", apiClient.twin_id, yggIP)
+		return errors.Wrapf(err, "sent request but didn't arrive to me. make sure the twin id is associated with a valid yggdrasil ip, twin id: %d, ygg ip: %s, err", threefoldPluginClient.twinID, yggIP)
 	}
 	return nil
 }
 
-func validateRMB(apiClient *apiClient, sub subi.SubstrateExt) error {
-	if err := validateRedis(apiClient); err != nil {
+func validateRMB(threefoldPluginClient *threefoldPluginClient, sub subi.SubstrateExt) error {
+	if err := validateRedis(threefoldPluginClient); err != nil {
 		return err
 	}
-	if err := validateYggdrasil(apiClient, sub); err != nil {
+	if err := validateYggdrasil(threefoldPluginClient, sub); err != nil {
 		return err
 	}
+
 	return nil
 }
 
-func validateRMBProxyServer(apiClient *apiClient) error {
-	return apiClient.grid_client.Ping()
+func validateRMBProxyServer(threefoldPluginClient *threefoldPluginClient) error {
+	return threefoldPluginClient.gridProxyClient.Ping()
 }
 
-func validateRMBProxy(apiClient *apiClient) error {
-	if err := validateRMBProxyServer(apiClient); err != nil {
+func validateRMBProxy(threefoldPluginClient *threefoldPluginClient) error {
+	if err := validateRMBProxyServer(threefoldPluginClient); err != nil {
 		return err
 	}
 	return nil
 }
 
-func preValidate(apiClient *apiClient, sub subi.SubstrateExt) error {
-	if apiClient.use_rmb_proxy {
-		return validateRMBProxy(apiClient)
+func validateMnemonics(mnemonics string) error {
+	if len(mnemonics) == 0 {
+		return errors.New("Mnemonics required")
+	}
+
+	alphaOnly := regexp.MustCompile(`^[A-Za-z]+$`)
+	if !alphaOnly.MatchString(mnemonics) {
+		return errors.New("Mnemonics are can only be composed of a non-alphanumeric character or a whitespace.")
+	}
+
+	return nil
+}
+
+func validateClientRMB(threefoldPluginClient *threefoldPluginClient, sub subi.SubstrateExt) error {
+	if threefoldPluginClient.useRmbProxy {
+		return validateRMBProxy(threefoldPluginClient)
 	} else {
-		return validateRMB(apiClient, sub)
+		return validateRMB(threefoldPluginClient, sub)
 	}
 }
 
```

Comment:
I know it's not part of the PR,but can we while we are at it change `validateAccountMoneyForExtrinsics` to be `validateAccountBalanceForExtrinsics`, balance is more accurate than money